numpy/lib/histograms.py

```diff
@@ -349,6 +379,41 @@ def _unsigned_subtract(a, b):
         # signed to unsigned
         return np.subtract(a, b, casting='unsafe', dtype=dt)
 
+def _get_auto_bin_edges(a, n_equal_bins, first_edge, last_edge):
+    """
+    Computes the bins used internally by `np.histogram(bins='auto')`.
+
+    Parameters
+    ==========
+    a : ndarray
+        Ravelled data array
+    first_edge, last_edge
+        Forwarded arguments from `_get_outer_edges`.
+    n_equal_bins : ndarray, optional
+        Ravelled weights array, or None
+
+    Returns
+    =======
+    bin_edges : ndarray
+        Array of bin edges
+    """
+    keep = (a >= first_edge)
+    keep &= (a <= last_edge)
+    if not np.logical_and.reduce(keep):
+        a = a[keep]
+
+    if a.size == 0:
+        return np.array([0,1])
+
+    width = np.ptp((first_edge,last_edge))/(n_equal_bins)
+    left_edges = np.floor_divide((a - first_edge), width) * width
```

Comment:
can width ever be 0?

Reply:
For the width calculation, `first_edge` and `last_edge` are forwarded arguments from `_get_outer_edges`. There are several checks in that function that would prevent `np.ptp((first_edge,last_edge))` from being 0. For example, if `first_edge` and `last_edge` are found to be equal in `_get_outer_edges`, then just before it returns their values, it performs this expansion:

```
# expand empty range to avoid divide by zero
    if first_edge == last_edge:
        first_edge = first_edge - 0.5
        last_edge = last_edge + 0.5
```

I think it is safe to say that width cannot be zero here.